pkg/machinery/config/types/v1alpha1/v1alpha1_types.go

```diff
@@ -867,11 +869,52 @@ type MachineDisk struct {
 	DiskPartitions []*DiskPartition `yaml:"partitions,omitempty"`
 }
 
+// DiskSize partition size in bytes.
+type DiskSize uint64
+
+// MarshalYAML write as human readable string.
+func (ds DiskSize) MarshalYAML() (interface{}, error) {
+	if ds%DiskSize(1000) == 0 {
+		bytesString := humanize.Bytes(uint64(ds))
+		// ensure that stringifying bytes as human readable string
+		// doesn't lose precision
+		parsed, err := humanize.ParseBytes(bytesString)
```

Comment:
I know that encode/decode seems weird, but that's the most robust way to ensure that it's safe to use humanize library for a value.
If that doesn't look great we can also reimplement this method by ourselves.